### ISSUE TITLE
Accessibility (a11y) issues/audit

### ISSUE THREAD

**Issue by Mythra** [vcpkg/vcpkg.github.io#19]
Congrats on getting to a beta point! It seems like no one has done a screenreader, and other a11y audits yet. So as a screenreader user myself I ran into some things I wanted to mention with my 30 seconds, but ideally much more indepth audit is conducted for people with other disabilities as well.

Below I'll try to document each of the 4 things I personally noticed in my very quick peek, why I find them important, a link to the wcag docs about them, and proposed solutions. However, if anything is unclear please don't hesitate to ask! (Also I file them as one issue for me, but if you want to close this , and split it up one into each issue, and then one for a final sweeping audit that's more than fine, I'm happy to follow those too!)

---

## Landmark Regions ##

***WCAG information***: https://www.w3.org/TR/WCAG20-TECHS/ARIA11.html

***Why this is useful to me***:

  Similar to how a user can jump around visually with their eyes, landmark regions really help me jump to sections of the page I care about, and navigate effectively. Imagine if you read halfway through a page, and you hear something interesting. Wanna go to the page that talks about that thing but to do so , you have to speed through all the elements you just read backwards until you hear something you know is in the navigation.

  Now compare that to just hearing the thing you find interesting, jumping to navigation, and then navigating to the option in the navigation you care about.

  One of those is a lot easier, and makes me actually want to browse the site. The other adds a lot of overhead making the docs a pain to navigate.

***What I'd expect***:

  I'd normally expect to see three landmark regions. Two navigation (one for top, one for side nav), and one main navigation landmark. You can use `role=""` attributes on elements, but these days it's usually recommended to just use the elements that are named: `<nav>`, and `<main>` which are html5 elements that implicitly add the correct roles.

---

## Skip to Main Content Link ##

***WCAG Information***: https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G1

Why this is useful to me:

  This one is admittedly fairly similar to "Landmark Regions". This is another optimization that allows screenreader users to not have to listen to the top nav, the left nav, and then get to the main content. It also is generally quicker than having to open our rotor, and jump to the main landmark region. I can just "Tab" -> "Enter", and be off. Which is super useful when I'm linked a page and just want to read that page.

  Again for my personal use (though everyone is different), I generally jump backwards to main areas with landmark regions (or somewhere that isn't main), but expect "Tab" -> "Enter" to just sort of get me to what I care about the quickest.

***What I'd Expect***:

  My first tab on the page to take me to a link that is called "Skip to main content", that when I hit enter, jumps the page to the main content of the page.

  A couple notes/gotchas:

    - The skip to content link doesn't need to be visible on the page until someone hits tab. This is what GitHub, google search, etc. do.
    - Please don't try to be fancy with this. Just a link with: `a href="#main`, and then the main block having: `id="main"`, is the best way to do it. I've seen many attempts that try to be fancy with javascript and ultimately ship a broken experience.

---

## Documentation Top Nav Link Mouse only hover content ##

***WCAG Information*** (this was the closest I could find, I realize it's not the same notes, but hopefully still helpful): https://www.w3.org/WAI/tutorials/menus/flyout/

***Why this is useful***:

  I'd like to be able to get to the same content a sighted user can 😄 , don't feel this one needs that much explanation. 😅 

***What I'd Expect***:

  To be able with a screenreader, and just a keyboard to be able to open the submenu and be able to navigate all the links within.

---

## Side Navigation doesn't show "Focus Rings" ##

***WCAG Information*** (This seems the most applicable): https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

***Why this is important to me***:

  When navigating with my screenreader on the side navbar (for example on this page): https://vcpkg.io/en/docs/README.html I hear things like "Clickable - Users", "Clickable - Specifications", but there are no focus rings. This creates a visual disconnect for low sighted users like me where I hear things but can't quite tell where I am.

  I should also note that this is really indicative of problems with KO "keyboard only" navigation, which some folks with motor disabilities have to use. So it'd definitely be worth checking this at the same time.

***What I'd Expect***:

  To be able to with a screenreader, and only a keyboard (with no screenreader) to be able to fully navigate the side navigation like a side user.

**Comment by dan-shaw** [comment]
Thanks for the review! I have made the following changes:

1. Landmark Regions: added the `nav`, `main`, and `footer` elements.
2. Skip to Main: I added a `Skip to Main Content` link that appears if you type `Tab`. The first tab should display this message.
3. Documentation Nav Hover Only: I adjusted it so that if you `Tab` to the arrow next to the Documentation link, and press `Enter`, it should display the hover only links.
4. Side Navigation and Focus: I have changed the side list elements to HTML buttons, which can be focused by default.

Let me know if these fixes work or if I missed anything.

**Comment by Mythra** [comment]
Hey @dan-shaw ! Sorry it took me a second to get back to this. I was able to take a look at the things I checked (though someone should probably spend more time looking 😅 ) and this is some great work thank you!

Two quick notes though, the documentation button tab at the top:

  - reads as "Clickable Button", and is seperate from the "Documentation" link. This means if you've skipped to main content, and are now coming back up the page you'll hear "Clickable button" right after github as opposed to documentation (might be a bit confusing!) Perhaps we could give it a better label? With `aria-label` something like: `aria-label="Open Documentation Quicklink Menu"`? (the wording there probably isn't perfect 😅 )

  - When you click the button there is no announcement/focus move to indicate something opened (in fact it appears after the top nav in the dom so the screenreader if you try to tab to see if something opened will read "GitHub", and "Blog" before even getting to the new links that are now open). This can be confusing, perhaps we could announce the menu is now open, or just move focus to the first item in the submenu?

Thank you though for working on these so quickly ❤️ it really means a lot!
